test/functional/wallet_resendwallettransactions.py

```diff
@@ -2,31 +2,77 @@
+        self.tx_invs_received = defaultdict(int)
+
+    def on_inv(self, message):
+        # Store how many times invs have been received for each tx.
+        for i in message.inv:
+            if i.type == 1:
+                # save txid
+                self.tx_invs_received[i.hash] += 1
 
 class ResendWalletTransactionsTest(BitcoinTestFramework):
     def set_test_params(self):
         self.num_nodes = 1
-        self.extra_args = [['--walletbroadcast=false']]
 
     def skip_test_if_missing_module(self):
         self.skip_if_no_wallet()
 
     def run_test(self):
-        # Should raise RPC_WALLET_ERROR (-4) if walletbroadcast is disabled.
-        assert_raises_rpc_error(-4, "Error: Wallet transaction broadcasting is disabled with -walletbroadcast", self.nodes[0].resendwallettransactions)
+        node = self.nodes[0]  # alias
+
+        node.add_p2p_connection(P2PStoreTxInvs())
+
+        self.log.info("Create a new transaction and wait until it's broadcast")
+        txid = int(node.sendtoaddress(node.getnewaddress(), 1), 16)
+
+        # Can take a few seconds due to transaction trickling
+        def wait_p2p():
+            self.bump_mocktime(1)
+            return node.p2p.tx_invs_received[txid] >= 1
+        wait_until(wait_p2p, lock=mininode_lock)
+
+        # Add a second peer since txs aren't rebroadcast to the same peer (see filterInventoryKnown)
+        node.add_p2p_connection(P2PStoreTxInvs())
+
+        self.log.info("Create a block")
+        # Create and submit a block without the transaction.
+        # Transactions are only rebroadcast if there has been a block at least five minutes
+        # after the last time we tried to broadcast. Use mocktime and give an extra minute to be sure.
+        block_time = self.mocktime + 6 * 60
+        node.setmocktime(block_time)
+        block = create_block(int(node.getbestblockhash(), 16), create_coinbase(node.getblockchaininfo()['blocks']), block_time)
+        block.nVersion = 3
+        block.rehash()
+        block.solve()
+        node.submitblock(ToHex(block))
 
-        # Should return an empty array if there aren't unconfirmed wallet transactions.
-        self.stop_node(0)
-        self.start_node(0, extra_args=[])
-        assert_equal(self.nodes[0].resendwallettransactions(), [])
+        # Transaction should not be rebroadcast
+        node.p2ps[1].sync_with_ping()
+        assert_equal(node.p2ps[1].tx_invs_received[txid], 0)
 
-        # Should return an array with the unconfirmed wallet transaction.
-        txid = self.nodes[0].sendtoaddress(self.nodes[0].getnewaddress(), 1)
-        assert_equal(self.nodes[0].resendwallettransactions(), [txid])
+        self.log.info("Transaction should be rebroadcast after 30 minutes")
+        # Use mocktime and give an extra 5 minutes to be sure.
+        rebroadcast_time = self.mocktime + 41 * 60
+        node.setmocktime(rebroadcast_time)
+        self.mocktime = rebroadcast_time
+        def wait_p2p_1():
```

Comment:
15646: I messed up in my fix a bit it seems 🙈 
```
test/functional/wallet_resendwallettransactions.py:78:9: E306 expected 1 blank line before a nested definition, found 0
^---- failure generated from test/lint/lint-python.sh
```
```suggestion

        def wait_p2p_1():
```